homework2/task01/project1/task1.py

```diff
@@ -43,26 +39,27 @@ def get_rarest_char(file_path: str) -> str:
     """Return one single rarest char."""
     with open(file_path, "r", encoding="unicode_escape") as f:
         text = f.read()
-        di = {}
+        all_chars = {}
         for i in text:
-            if i not in di:
-                di[i] = 1
+            if i not in all_chars:
+                all_chars[i] = 1
             else:
-                di[i] += 1
-        for key, value in di.items():
-            if value == min(di.values()):
+                all_chars[i] += 1
+        min_value = min(all_chars.values())
+        for key, value in all_chars.items():
+            if value == min_value:
                 return key
```

Comment:
Can you find a key with minimum value without a second iteration?

Reply:
Fixed